src/main/java/core/basesyntax/strategy/BalanceOperation.java

```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class BalanceOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Map<String, Integer> inventory) {
```

Comment:
The `handle` method should ensure that the quantity being put into the inventory is not negative, as this would represent an invalid state of the inventory. It's important to validate input data to maintain the integrity of the system.

Reply:
Implement this comment.